### PR TITLE
First pass at upgrading to QP

### PR DESCRIPTION
**Pull request by alexolivier**
- [x] Basic eCommerce Spec
- [x] Transaction Template
- [x] ~~Handle Variant Change~~ Can't be added via theme - details in docs
- [x] Update implementation docs

### CODE REVIEW

File: qubit-events.liquid [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        productId: "{{ product.id }}",
+        name: {{ product.title | json }},
+        stock: {{ product.variants.first.inventory_quantity }},
+		manufacturer: {{ product.vendor | json}},
+        price: {
+        	currency: "{{ shop.currency }}",
+        	value: {{ product.price | divided_by: 100 }}
+     	 },
+      	url: "{{ product.url }}",
+     	description: {{ product.description | strip_html | truncate: 100 | json }},
```
Comment (anthonydelage) [inline qubit-events.liquid]: Why are we truncating at 100 characters? The QP `Text` type goes up to 2048 chars.

**Comment by anthonydelage** [comment]
Do we have a sandbox we can test this in?

File: qubit-events.liquid [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      	category: ["{{ product.type}}"],
```
Comment (anthonydelage) [inline qubit-events.liquid]: Is there any way to convert this to the new field, `categories`?

Comment (anthonydelage) [inline qubit-events.liquid]: Is there any way to convert this to the new field, `categories`?

**Comment by alexolivier** [comment]
Added @adelage to our dev site uv-dev.myshopify.com

File: qubit-events.liquid (hunk 2 above)
Comment (alexolivier) [reply qubit-events.liquid]: The shopify data is a product type and a set of tags

File: qubit-events.liquid (hunk 1 above)
Comment (alexolivier) [reply qubit-events.liquid]: Copied from old version

File: README.md [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
 
 #Overview
 
-When implementing [Universal Variable](http://tools.qubitproducts.com/uv) and [Opentag](http://www.qubitproducts.com/tag-management), you need to do two things from a technical perspective:
+When implementing Qubit, you need to do two things from a technical perspective:
 
-1. Add the Opentag container to the `<head>` of every page
-2. Add the Universal Variable (UV) directly before the Opentag container on every page.
+1. Add the Smartserve container to the `<head>` of every page
+2. Emit emits based on user browsing behaviour
```
Comment (anthonydelage) [inline README.md]: Does the UV API need to be added in the head, or are we assuming that Smartserve will load before the events start firing?

**Comment by alexolivier** [comment]
SS has aync on it so no guarantees so keep UV API

File: README.md (hunk 3 above)
Comment (alexolivier) [reply README.md]: SS has aync on it so no guarantees so keep UV API

File: qubit-events-transaction.liquid [hunk 4]
```diff
@@ -60,7 +60,7 @@
         }, 
         url: "{{ item.url }}",
         description: {{ item.product.description | strip_html | truncate: 2048 | json }},
-        category: [{{ item.product.type | json}}],
+        categories: [{{ item.product.type | json}}],
         images: [{{ item.image | json }}]
```
Comment (anthonydelage) [inline qubit-events-transaction.liquid]: @alexolivier is it actually `image` when it's an `item`, and `images` when it's a `product`?

File: README.md (hunk 3 above)
Comment (anthonydelage) [reply README.md]: I'd add a blurb about including the UV API script then, as per [our docs](https://docs.qubit.com/implementing-qp-ec.html).

File: qubit-events-transaction.liquid (hunk 4 above)
Comment (alexolivier) [reply qubit-events-transaction.liquid]: Looks like it in the Shopify API

File: README.md [hunk 5]
```diff
@@ -1,63 +1,87 @@
...
+    eventType: 'detail',
+    product: {
+      sku: {{ product.sku | default: product.variant_id | json }},
+      productId: "{{ product.product_id }}",
+      name: {{ product.title | json }},
+      manufacturer: {{ product.vendor | json}},
+      stock: {{ product.inventory_quantity }},
+      price: {
+        currency: "{{ shop.currency }}",
+        value: {{ product.price | divided_by: 100 }}
```
Comment (anthonydelage) [inline README.md]: Is the standard price format multiplied by 100? We might want to use a [money filter](https://help.shopify.com/themes/liquid/filters/money-filters) here instead.

File: qubit-events-transaction.liquid [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      value: {{ checkout.total_price | money_without_currency }}
+    },
+    total: {
+      currency: '{{shop.currency}}',
+      value: {{ checkout.total_price | money_without_currency }}
+    }
+  }
+
+  var transactionSummary = {
+    id: "{{ order_number }}"
```
Comment (anthonydelage) [inline qubit-events-transaction.liquid]: Any reason why we don't have the `checkout.` prefix here?

**Pull request merged**